File: 01-FUNDAMENTOS_DE_PROGRAMACAO/C/Exercicio 18.c
/* 
18 - Escreva um programa em que leia 3 cadeias de caracteres de tamanho máximo 50, e as imprime na tela em ordem alfabética.
*/

#include<stdio.h>
#include<string.h>
#include<locale.h>

int main(){
  setlocale(LC_ALL, "Portuguese");
  char nm[3][20], aux[20];
  int i, j;

  for (i = 0; i < 3; i++) {
    printf("\nNome %i: ", i+1);
    scanf("%s", nm[i]);
  }

  for (i = 1; i < 3; i++) { /* 3 = qtde de palavras */
    for (j = 1; j < 3; j++) {
      // verifica se tem que ser depois, se for troca de posição
      if (strcmp(nm[j - 1], nm[j]) > 0) {
          strcpy(aux, nm[j - 1]);
          strcpy(nm[j - 1], nm[j]);
          strcpy(nm[j], aux);
      }
    }
  }

  for (i = 0; i < 3; i++)
    printf("\n%s", nm[i]);
}

File: 01-FUNDAMENTOS_DE_PROGRAMACAO/Java/Exercicio18.java
/*
18 - Escreva um programa em que leia 3 cadeias de caracteres de tamanho máximo 50,
e as imprime na tela em ordem alfabética.
 */

import java.util.Arrays;
import java.util.Scanner;

public class exercicio18 {
    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        
        String vetor[] = new String[3];
        for(int i=0; i<3; i++){
            System.out.println("Nome "+i+": ");
            vetor[i] = teclado.next();
        }
        
        //com o .sort você ordena o vetor
        Arrays.sort(vetor);

        //Impressão
        System.out.print ("\nO nomes em ordem alfabética são \n");
        for (int i = 0 ; i<vetor.length; i++)
            System.out.print (vetor[i]+", \n");
    }    
}
